src/models/commands.ts

```diff
@@ -319,25 +327,28 @@ export const fcCommands: FCCommand[] = [
         commandName: 'ForceCode.refreshContext',
         name: 'Retrieving ',
         hidden: true,
-        command: function (context, selectedResource?) {
+        command: async function (context, selectedResource?) {
             if(selectedResource && selectedResource instanceof Array) {
-                var files: ToolingType[] = [];
-                selectedResource.forEach(curRes => {
-                    var tType: string = getAnyTTFromFolder(curRes);
-                    var index: number = getTTIndex(tType, files);
-                    const theName: string = getAnyNameFromUri(curRes);
-                    if(index >= 0) {
-                        if(theName === '*') {
-                            files[index].members = ['*'];
-                        } else {
-                            files[index].members.push(theName);
-                        }
-                    } else {
-                        files.push({name: tType, members: [theName]});
-                    }
+                return new Promise((resolve) => {
+                    var files: PXMLMember[] = [];
+                    var proms: Promise<PXMLMember>[] = selectedResource.map(curRes => getAnyNameFromUri(curRes));
+                    Promise.all(proms).then(theNames => {
+                        theNames.forEach(curName => {
+                            var index: number = getTTIndex(curName.name, files);
+                            if(index >= 0) {
+                                if(curName.members === ['*']) {
+                                    files[index].members = ['*'];
+                                } else {
+                                    files[index].members.push(...curName.members);
```

Comment:
I would prefer `return`ing a `map` instead of `forEach`ing over a list and pushing a value to an array.